src/logging.cpp

```diff
@@ -443,3 +489,22 @@ void BCLog::Logger::ShrinkDebugFile()
     else if (file != nullptr)
         fclose(file);
 }
+
+bool BCLog::Logger::SetLogLevel(const std::string& level_str)
+{
+    const auto level = GetLogLevel(level_str);
+    if (!level || level > BCLog::Level::Info) return false;
+    m_log_level = level.value();
+    return true;
+}
+
+bool BCLog::Logger::SetCategoryLogLevel(const std::string& category_str, const std::string& level_str)
+{
+    BCLog::LogFlags flag;
+    if (!GetLogCategory(flag, category_str)) return false;
+
+    const auto level = GetLogLevel(level_str);
+    if (!level || level > BCLog::Level::Info) return false;
```

Comment:
Should this be `DEFAULT_LOG_LEVEL` instead of `BCLog::Level::Info`?

Right now they are the same, but if `DEFAULT_LOG_LEVEL` is changed, this should be changed too, right? If "yes", then this should be `DEFAULT_LOG_LEVEL`.

Reply:
No, I considered this but they don't share the exact same logic; the default level isn't necessarily linked to the unconditional level.

Reply:
Hoisted to a `MIN_USER_SETABLE_LEVEL` constant.